Refactoring (především WsdlManager)

xificurk opened the pull request:
Prošel jsem `WsdlManager`, snad jsou teď jednotlivé metody trochu čitelnější.
Z třídy `Skautis` byla odstraněna explicitní závislost na `Config` - pro těch pár věcí si šahá do `WsdlManageru`. (Tak jak to bylo doteď to stejně moc nefungovalo - config ve `Skautis` a `WsdlManager` se mohl lišit a to by asi nedělalo dobrotu.)

xificurk committed:
Přejmenování WS -> WebService v názvu metod.

xificurk committed:
WsdlManager refactoring
- přejmenování metod a proměnných na čitelnějí tvary
- úklid v kódu a doplnění PHPdoc bloků

xificurk committed:
Fix unit tests

xificurk committed:
Odstraněna volba $profiler z Config a přepracováno jednoduché logování.
- Pro pokročilejší debugger/profiler je vhodnější zaregistrovat
  vlastní listener na WsdlManageru.
- Pro jednoduchý debugging je možné použít metody enableDebugLog() a
  getDebugLog() přímo na Skautis objektu.

xificurk committed:
Veškerá logika získávání objektu webové služby přesunuta do WsdlManager.
(+vyřešen problém se změnou bezpečnostního tokenu)

xificurk committed:
Skautis: odstraněna explicitní závislost na Config

JindrichPilar commented:
logovani soap callu bych rad nejakym zpusobem zachoval, at uz to bude logovat cokoliv. Podstatne je aby clovek co nema poneti jak funguji event listenery byl schopen si vypsat requesty a debugovat.
Z jeho pohledu linearne:

```
ANO CHCI DEBUGOVAT  -> $profiler = true;
......
TAK MI DEJ VYSLEDKY -> $skautis->*****->getQueries();
```

Pokud by tam to logovani nebylo, klidne by se mohl vypustit `$profiler` a volat `dispatch` vzdy. (clovek si to debugovani zapne tim ze registruje listener). Coz by nebylo spatne tak jako tak.

EDIT: I kdyby to byla pridana externi vec v HelperTrait

xificurk commented:
Dispatch se už volá vždy (když je definován listener). `$profiler` bych z `Config` každopádně vypustil (to jsem nějak zapomněl).
Ale máš pravdu, že se hodí mít nějaké jednoduché logování přímo na `Skautis` objektu, takže tam něco přidám a pushnu sem návrh řešení.

xificurk committed:
Config:
- Odstraněna nepoužívaná metoda isSetAppId().
- Odstraněna metoda validate - objekt Config v nevalidním stavu je k
  ničemu a tedy není důvod dovolit jeho vznik. Pokud není specifikováno
  $appID vyhazuje konstruktor výjimku, typovost ostatní boolean
  argumentů není kontrolována, ale setter zadanou hodnotu přetypovává.
- Doplněny PHPdoc bloky.
- TestMode getter přejmenován na trochu čitelnějěí isTestMode() a setter
  má nyní výchozí hodnotu true.

xificurk committed:
Skautis:
- Opraveno escapování generovaných URL.
- Zrušen zbytečný zápis do session v konstruktoru.
- U časů vrácených ze skautISu se předpokládá, že jsou v TZ
  Europe/Prague.
- Doplněny PHPdoc bloky a sjednocen coding style.

xificurk commented:
Tak za mě asi takhle prozatím vše.
Až/jestli to mergneš, tak bych se podíval na aktualizaci Nette extension.

JindrichPilar commented:
Pokud bys jeste mohl zanest zmeny do [CHANGELOG.md](https://github.com/skaut/Skautis/blob/2.x/CHANGELOG.md). Jsou to zatim zmeny vuci 1.x, takze klidne smazat/upravit co uz neplati po tve uprave. Staci jednoduche poznamky typu:

> WS prejmenovan na WebService

xificurk commented:
ok, doplním... někdy v následujících dnech. I když teda po tvém rozdělní a mém řádění s lepším pojmenováním to znamená, že se v podstatě kompletně změnilo API :-)
A až se trochu usadí změny, tak pomůžu i s dokumentací.

sinacek commented:
Ahoj,
díky moc oběma za vaši činnost, nikdy mě nenapadlo, že se ta knihovna takle posune! Jinak bych byl klidně pro nedělat changelog, ale jen pak napsat návod, jak přejít z verze 1.x na 2.x a to bude podle mě dostačující, protože nevím o nikom, kdo by se pouštěl do nějakého vnitřního přepisování. Tedy bych dělal pak changelog až v rámci verze 2.x

JindrichPilar commented:
Jak psal @xificurk navod bude dobre sepsat az se zmeny usadi. To jest nekdy pred vydanim, aby byl co nejaktualnejsi.  
Psat changelog  1.x -> 2.x mi prijde dobre i z pohledu ze kdyz prijde nekdo novy, staci mu kouknout do jednoho souboru a vi jestli je knihovna aktivni, pouzite design paterny, kterym smerem se pohybuje a odkud jde.

sinacek commented:
Návod určitě až potom, jen mi příjde, že těch změn je opravdu hodně a to jak obsahově, tak názvy, atd. ale klidně to být může, to není nic proti ničemu
